table.go

```diff
@@ -1216,8 +1257,12 @@ func (t *TableBlock) Size() int64 {
 }
 
 // Index provides atomic access to the table index.
-func (t *TableBlock) Index() *btree.BTree {
-	return (*btree.BTree)(t.index.Load())
+func (t *TableBlock) Index() (*index, func()) {
+	index := (*index)(t.index.Load())
+	index.ref.Add(1)
+	return index, func() {
```

Comment:
I wonder if a function bound to the `index` would be more intuitive. like `.release()`? don't feel strongly though... I think I understand what you were doing though, since it's part of the return value it's hard to misuse/ignore